### ISSUE TITLE
Azblob: DownloadBlobToWriterAt gets context cancelled

### ISSUE THREAD

**Issue by eccles** [Azure/azure-sdk-for-go#17670]
### Bug Report
import azStorageBlob "github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
0.3.0
go version go1.17.7 linux/amd64

Attempts to download a blob with the following call:

```
func (rr *ReaderResponse) DownloadToWriter(ctx context.Context, w io.Writer) error {
        return rr.BlobClient.DownloadBlobToWriterAt(
                ctx,   /// context.BackGround() in most cases
                0,
                rr.CountToEnd,    /// standard MS constant
                newWriterAt(w),   /// convrets io.Writer to io.WriterAt with zero offset
                azStorageBlob.HighLevelDownloadFromBlobOptions{},
        )
}
```

when calling using a http.ResponseWriter as the io.Writer results in a context cancelled error **most** of the time.
About 1 in 4 attempts are successful.

The error is "copied to out with errors: context canceled"

Expected:
    No context cancelled error and successful download

Similar code using the older azure-storage-blob-go repo worked. This was discovered whilst moving across to azure-sdk-for-go.

**Comment by eccles** [comment]
Note that various ctx args were tried such as context.Background() to no avail

**Comment by eccles** [comment]
I have reworked the code to get the Body and feed that as a Reader to io.Copy.

`       resp.BlobClient = azp.containerClient.NewBlobClient(identity)
        resp.CountToEnd = int64(azStorageBlob.CountToEnd)
        get, err := resp.BlobClient.Download(
                ctx,
                &azStorageBlob.DownloadBlobOptions{
                        BlobAccessConditions: &resp.blobAccessConditions,
                        Count:                &resp.CountToEnd,
                },
        )
        if err != nil && err == io.EOF { // nolint
                logger.Sugar.Infof("cannot get blob body: %v", err)
                return nil, ErrorFromError(err)
        }
        resp.Reader = get.Body(&azStorageBlob.RetryReaderOptions{})
       _, err = io.Copy(w, respReader)
        if err != nil {
                logger.Sugar.Infof("copied to out with errors: %v", err)
        }

**Comment by SaurabhSharma-MSFT** [comment]
@eccles We are looking into it and get back to you for any additional information.

**Comment by eccles** [comment]
thank you for your attention - this is not urgent but will require a fix if we are to move to the new azure-sdk as recommended by yourselves.

**Comment by jhendrixMSFT** [comment]
This might have been due to an issue in `azcore` which has since been fixed.  https://github.com/Azure/azure-sdk-for-go/pull/17797

We've just released an updated version `azblob@v0.5.0` that contains this (and other fixes). Note that we've refactored the public surface area. However, we feel that it makes navigating the package much easier.  Can you please try it and report back?
